serverless/src/functions/apps/handler.ts

```diff
@@ -0,0 +1,106 @@
+import { APIGatewayProxyResult, APIGatewayEvent } from 'aws-lambda';
+import { formatJSONResponse } from '@libs/api-gateway';
+import * as dotenv from 'dotenv';
+import { v4 } from 'uuid';
+import { prisma } from '@libs/prisma';
+import {
+  BunnyCdn,
+  BunnyCdnError,
+  CreatePullZoneMethodArgs,
+} from '@libs/bunnyCDN';
+
+export const submitAppInfo = async (
+  event: APIGatewayEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    // Check the parameters and environment variables
+    dotenv.config();
+    if (event.body === null || process.env.BUNNY_CDN_ACCESS_KEY == undefined) {
+      return formatJSONResponse({
+        status: 422,
+        message: 'Required parameters were not passed.',
+      });
+    }
+
+    // Set up constants
+    const bunnyCdn = new BunnyCdn(process.env.BUNNY_CDN_ACCESS_KEY);
+    const data = JSON.parse(event.body);
+    const appInfo = {
+      apId: 'null',
+      createdAt: new Date().toISOString(),
+      sourceDomain: data.sourceDomain,
+      hostname: data.targetDomain,
+    };
+
+    let callSuccess;
+    let pullZone: {
+      id: any;
+      name?: string;
+      originUrl?: string;
+      hostname?: string;
+    };
+
+    do {
+      let id = v4();
+      let requestArgs: CreatePullZoneMethodArgs = {
+        zoneId: id, // this is technically the zone name. It should be unique.
+        originUrl: appInfo.sourceDomain,
+      };
+
+      try {
+        pullZone = await bunnyCdn.createPullZone(requestArgs);
+        appInfo.apId = id;
+        callSuccess = true;
+      } catch (error) {
+        callSuccess = false;
+        if (
+          error instanceof BunnyCdnError &&
+          error.name === 'pullzone.name_taken'
+        ) {
+          continue;
+        } else {
+          throw error;
+        }
+      }
+    } while (callSuccess === false);
```

Comment:
I understand that this is not expected to, but may result into infinite loop. Maybe would be a good idea create max number of tries exception or something like that.

Reply:
good point